src/components/vr_cooperation/src/base_command_request.cc

```diff
@@ -187,81 +136,47 @@ bool BaseCommandRequest::ParseMobileResultCode(const Json::Value& value,
   }
 
   result_code = GetHMIResultCode(mobile_result_code);
-  if (IsMember(value[kError], kMessage)) {
-    info = value[kError][kMessage].asString();
-  }
-
-  // TODO (KKarlash) compare result_code with HMI result code values
-
-  return result_code != -1;
 }
 
-void BaseCommandRequest::PrepareRequestMessageForMobile(const char* function_id,
-    const Json::Value& message_params,
+void BaseCommandRequest::PrepareRequestMessageForMobile(
+    vr_hmi_api::RPCName function_id,
+    const std::string& message_params,
     application_manager::MessagePtr& message) {
   LOG4CXX_AUTO_TRACE(logger_);
   Json::Value msg;
   msg[kId] = service_->GetNextCorrelationID();
   msg[kJsonrpc] = kJsonRpc;
   msg[kMethod] = function_id;
-  if (!message_params.isNull()) {
+  if (!message_params.empty()) {
     msg[kParams] = message_params;
```

Comment:
Not sure about this param, based on RSDLmobileAPI.xml ActivateService request for mobile should include attributes such as: "service", "success", "resultCode", "info"